src/Spice86/Spice86DependencyInjection.cs

```diff
@@ -199,27 +196,30 @@ internal Spice86DependencyInjection(Configuration configuration, MainWindow? mai
             loggerService.Information("Function handler in external interrupt created...");
         }
 
-        Cpu cpu = new(interruptVectorTable, stack,
-            functionHandler, functionHandlerInExternalInterrupt, memory, state,
-            dualPic, ioPortDispatcher, callbackHandler, emulatorBreakpointsManager,
-            loggerService, executionFlowRecorder);
+        ICPU cpu;
+        Func<UIDispatcher, ICpuViewModel> cpuViewModelSupplier;
 
-        if (loggerService.IsEnabled(LogEventLevel.Information)) {
-            loggerService.Information("CPU created...");
-        }
+        if (!configuration.CfgCpu) {
+            cpu = new Cpu(interruptVectorTable, stack,
+                functionHandler, functionHandlerInExternalInterrupt, memory, state,
+                dualPic, ioPortDispatcher, callbackHandler, emulatorBreakpointsManager,
+                loggerService, executionFlowRecorder);
 
-        CfgCpu cfgCpu = new(memory, state, ioPortDispatcher, callbackHandler,
-            dualPic, emulatorBreakpointsManager, functionCatalogue, loggerService);
+            if (loggerService.IsEnabled(LogEventLevel.Information)) {
+                loggerService.Information("CPU created...");
+            }
 
-        if (loggerService.IsEnabled(LogEventLevel.Information)) {
-            loggerService.Information("CfgCpu created...");
-        }
+            cpuViewModelSupplier = uiDispatcher => new CpuViewModel(state, memory, pauseHandler, uiDispatcher);
+        } else {
+            var cfgCpu = new CfgCpu(memory, state, ioPortDispatcher, callbackHandler,
+                dualPic, emulatorBreakpointsManager, functionCatalogue, loggerService);
+            cpu = cfgCpu;
 
-        IInstructionExecutor instructionExecutor = configuration.CfgCpu ? cfgCpu : cpu;
```

Comment:
Don't replace direct dependencies with indirect dependencies through the ICPU interface.